%YAML:1.0

camera:
  type: "PINHOLE"

  width: 640
  height: 480

  fx: 441.754
  fy: 441.452
  cx: 321.219
  cy: 239.310

  k1: -0.04020913586144525
  k2: 0.06694483918329086
  d1: -0.0006286858094548
  d2: 0.0002594613144265055

Tcb: !!opencv-matrix
  rows: 3
  cols: 4
  dt: d
  data: [ 0.01271782, -0.99991552,  0.00268397,  0.00294788,
         -0.99989602, -0.01269923,  0.00683216,  0.04075775,
         -0.00679749, -0.00277058, -0.99997306, -0.00963677]

track:
  verbose: 0
  fm_threshold: 1.0
  max_cornor_num: 300
  min_cornor_gap: 15
  track_frequency: 10 

initial:
  verbose: 1
  imu_cnt: 200
  imu_accl_cov: 0.1
  imu_gyro_cov: 0.01

msckf:
  verbose: 0
  noise_accl: 0.0025
  noise_gyro: 0.000025
  noise_accl_bias: 0.0001
  noise_gyro_bias: 0.000001
  noise_observation: 0.0001225

  ikf_iters: 10
  sliding_window_lens: 20
  distance_threshold: 0.5
  angle_threshold: 15 # in deg
  track_rate_threshold: 0.5

sfm:
  verbose: 0
  max_iter_cnt: 20
  max_try_cnt: 10
  converge_threshold: 5.e-7
  min_disparity_angle: 5 # in deg
  min_disparity_distance: 0.1


# save trajectory
save_trajectory: 0
save_path: "/home/ubuntu/catkin_ws/src/learn_msckf/results/trajectory.txt"
